src/pure.js

```diff
@@ -25,42 +18,83 @@ configureDTL({
   },
 })
 
+if (typeof React.startTransition !== undefined) {
+  configureDTL({
+    unstable_advanceTimersWrapper: cb => {
+      return act(cb)
+    },
+    asyncWrapper: cb => cb(),
```

Comment:
Is this necessary to configure?

Reply:
TL;DR: I misunderstood the question. I think we don't need to configure that because it's the default. I still leave my original long winded explainer for future reference and because it made me realize I still don't fully understand why `act()` behaves the way it does.

So let's say we have 

```js
waitFor(() => expect(getByRole('button')).toBeInTheDocument())
```
and the button would only appear after two intervals. This would previously translate to 
```js
act(() => {
   jest.advanceTimersByTime();
   checkCallback();
   jest.advanceTimersByTime();
   checkCallback();
});
```
The problem is that React only flushes effects when exiting the `act` call. So `checkCallback` could not assert on any effect scheduled due to timers being advanced.
If we just configure `unstable_advanceTimersWrapper` to use `act` we would now have
```
act(() => {
   jest.advanceTimersByTime();
   act(() => {
     checkCallback();
   })
   jest.advanceTimersByTime();
   act(() => {
     checkCallback();
   });
});
```
The problem now is that React only flushes scheduled effects in the outermost act so we need to remove that.

I think I good question is why only the outermost `act` is considered.

Reply:
I need to double check something though. React 17 will no longer have `act` as the `asyncWrapper` but somehow our React 17 tests are still passing.

Reply:
> I need to double check something though. React 17 will no longer have `act` as the `asyncWrapper` but somehow our React 17 tests are still passing. So either our tests are incomplete or this was dead code.

Did you mean React 18? If so, following [this comment](https://github.com/reactwg/react-18/discussions/23#discussioncomment-1087260) I'm not sure it won't work, it's just not the recommendation, isn't it?

Reply:
> Did you mean React 18? If so, following this comment I'm not sure it won't work, it's just not needed anymore, isn't it?

No, 17. I was reffering to our usage though. I changed the behavior for React 17 as well but no tests started failing.

Reply:
> I need to double check something though. React 17 will no longer have `act` as the `asyncWrapper` but somehow our React 17 tests are still passing. So either our tests are incomplete or this was dead code.

Tests are incomplete due to usage of class components. Backported the new tests with a function component in https://github.com/testing-library/react-testing-library/pull/962 that should ensure `asyncWrapper` is still present in React 17.

Reply:
Prevented mild disaster since the new tests also uncovered silly lines such as `typeof React.startTransition !== undefined`. It's either `React.startTransition !== undefined` or `typeof React.startTransition !== 'undefined'`.

Also uncovered that the "configure wrapper" approach is not compatible with supporting different timers across multiple versions of react.

Reply:
Going to leave this a bit until the next WG meeting. Unclear what the timeline is for work on act. Feels like the core team already moved on but there are still glaring holes in the behavior of act().

Reply:
Though I'm personally fine with shipping this as an alpha to test integration. We haven't solved all problems yet for the alpha but neither has React itself. I'm fine with shipping this in an incomplete state since I can just blame React about all the missing parts 😎

Reply:
Sounds fine to me 👍